### PR TITLE
contribute file to guide newbies 

### PR DESCRIPTION
**Pull request by GregoryKutuzov**

### CODE REVIEW

**Comment by GregoryKutuzov** [comment]
check it out and add rules I've been missed.

File: CONTRIBUTE.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+To contribute to this project you must obey the following rules:
```
Comment (VeselovAlex) [inline CONTRIBUTE.md]: Add header

File: CONTRIBUTE.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        //action
+        
+        //assert
+    }
+```
+
+#### implementation specific
+
+1. Use registers for unique name generation to guaranty that your tests are not overlapping with others.
+2. Use special safe check methods for arrange step.
```
Comment (VeselovAlex) [inline CONTRIBUTE.md]: jUnit-like assertions?

File: CONTRIBUTE.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    3. After your commit has been approved rebase your remote branch on actual master with a force push. 
+
+
+#### code style
+Use [standard](http://www.oracle.com/technetwork/java/codeconventions-150003.pdf) java code style.
+1. Each test must contain javadoc before test method declaration with a related issue number or a comment 
+that this test was added directly bypassing corporate issue tracker.
+```java
+    /**
+     * #1234
```
Comment (VeselovAlex) [inline CONTRIBUTE.md]: Alt. @issue 1234

**Pull request merged**